core/modules/layout/includes/block.text.inc

```diff
@@ -91,6 +91,18 @@ class BlockText extends Block {
           ),
         ),
       );
+      $form['convert']['description'] = array(
+       '#type' => 'textfield',
+       '#title' => t('Block description'),
+       '#maxlength' => 128,
+       '#description' => t('Used in administrative interfaces and not shown to end-users.'),
+       '#weight' => 3,
```

Comment:
@klonos I am a bit puzzled by your screenshot here because it is showing the 'Internal Name' field whereas in my testing this is hidden from the user and the 'machine name' is generated automatically. Perhaps this is some javascript.

Here is what I see before applying the PR
![screenshot-www ctwm org uk-2017-06-12-07-47-32](https://user-images.githubusercontent.com/10559274/27022739-39170daa-4f47-11e7-9290-0632f412fa39.png)

And here is what I see after applying the PR
![screenshot-backdrop margaretglass com-2017-06-12-07-50-38](https://user-images.githubusercontent.com/10559274/27022767-5772b970-4f47-11e7-8db2-b5f64a14d1eb.png)